specs/agents/tracing-spans-destination.md

```diff
@@ -100,13 +100,39 @@ exit = exit || context.destination || context.db || context.message || context.h
 
 If no value is set to the `context.destination.service.resource` field, the logic for automatically inferring 
 it MUST be the following:
+
+OPTION I:
 ```groovy
-if      (context.db?.instance)         "${subtype ?: type}/${context.db?.instance}"
+if      (context.db?.instance)         "${subtype ?: type}/${context.db.instance}"
+else if (context.db)                   subtype ?: type
 else if (context.message?.queue?.name) "${subtype ?: type}/${context.message.queue.name}"
-else if (context.http?.url)            "${context.http.url.host}:${context.http.url.port}"
+else if (context.message)              subtype ?: type
+else if (context.http?.url?.port > 0)  "${context.http.url.host}:${context.http.url.port}"
+else if (context.http?.url?.host)      context.http.url.host
 else                                   subtype ?: type
```

Comment:
Shouldn't grpc have its own branch? AFAIK the java agent sets `context.destination.service.resource`, but other agents might not do that.

Reply:
Maybe one could use to `context.destination.address` (if it exists) before using type/subtype

Reply:
This is defined in the [gRPC spec](https://github.com/elastic/apm/blob/master/specs/agents/tracing-instrumentation-grpc.md#span-context), so all agents should implement it.
This spec is about auto-inference of the resource field when it is not set otherwise.

Reply:
Ah, ok.